src/neo/SmartContract/Native/Votes/VoteContract.cs

```diff
@@ -0,0 +1,428 @@
+            if (pairs.Count() == 0) return false;
+            MultiStatistic result = new MultiStatistic();
+            List<UInt160> voteList = new List<UInt160>();
+            foreach (var pair in pairs)
+            {
+                VoteState vote_state = Neo.IO.Helper.AsSerializable<VoteState>(pair.Item2.Value);
+                UInt160 voter = vote_state.GetVoter();
+                if (voteList.Contains(voter)) continue;
+                voteList.Add(voter);
+                int account_balance = (int)new NeoToken().BalanceOf(engine.Snapshot, voter);
+                MultiCandidate candidate = Neo.IO.Helper.AsSerializable<MultiCandidate>(vote_state.ToArray());
+                result.AddVote(new CalculatedMultiVote(account_balance, candidate.GetCandidate()));
+            }
+            result.CalculateResult(new SchulzeModel());
+            var resultMatrix = result.ShowResult();
+            if (!AddResult(engine.Snapshot, resultMatrix, id))
+            {
+                return false;
+            }
+            return resultMatrix;
+        }
+
+        [ContractMethod(0_01000000, ContractParameterType.ByteArray)]
+        private StackItem GetSingleStatistic(ApplicationEngine engine, VMArray args)
+        {
+            if (args[0] is null) return false;
+            UInt256 TxHash = new UInt256(args[0].GetSpan());
+            var id = Crypto.Hash160(VoteCreateState.ConcatByte(TxHash.ToArray(), (engine.CallingScriptHash ?? this.Hash).ToArray()));
+            StorageKey create_key = CreateStorageKey(Prefix_CreateVote, id.ToArray());
+            StorageItem create_byte = engine.Snapshot.Storages.TryGet(create_key);
+            VoteCreateState createState = Neo.IO.Helper.AsSerializable<VoteCreateState>(create_byte.Value);
+            if (createState.IsSequence) return false;
+            if (!InteropService.Runtime.CheckWitnessInternal(engine, createState.Originator)) return false;
+            StorageKey index_key = CreateStorageKey(Prefix_Vote, id.ToArray());
+            byte[] prefix_key = StorageKey.CreateSearchPrefix(Hash, index_key.Key);
+            IEnumerable<(StorageKey, StorageItem)> pairs = engine.Snapshot.Storages.Find(prefix_key).OrderByDescending(p => p.Key);
+            if (pairs.Count() == 0) return false;
+            SingleStatistic result = new SingleStatistic();
+            List<UInt160> voterList = new List<UInt160>();
+            foreach (var pair in pairs)
+            {
+                VoteState vote_state = Neo.IO.Helper.AsSerializable<VoteState>(pair.Item2.Value);
+                UInt160 voter = vote_state.GetVoter();
+                if (voterList.Contains(voter)) continue;
+                voterList.Add(vote_state.GetVoter());
+                int account_balance = (int)new NeoToken().BalanceOf(engine.Snapshot, vote_state.GetVoter());
+                SingleCandidate candidate = Neo.IO.Helper.AsSerializable<SingleCandidate>(vote_state.GetCandidate().ToArray());
```

Comment:
I'm curious, is our use case voting for consensus nodes? 
Or can it be used for other dAPP. If it can be used for dAPP voting scenario, I think the most common is to count one vote per account. Do we need to support it?